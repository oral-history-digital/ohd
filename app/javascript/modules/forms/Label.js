import PropTypes from 'prop-types';
import classNames from 'classnames';

import { useI18n } from 'modules/i18n';

function Label({
    label,
    labelKey,
    mandatory,
    htmlFor,
    className,
}) {
    const { t } = useI18n();

    const l = label ? label : t(labelKey);

    return (
        <div className={classNames('form-label', className)}>
            <label className="FormLabel" htmlFor={htmlFor}>
                {`${l}${mandatory ? ' *' : ''}`}
            </label>
        </div>
    );
}

Label.propTypes = {
    label: PropTypes.string,
    labelKey: PropTypes.string,
    mandatory: PropTypes.bool.isRequired,
    htmlFor: PropTypes.string,
    className: PropTypes.string,
};

Label.defaultProps = {
    mandatory: false,
};

export default Label;
