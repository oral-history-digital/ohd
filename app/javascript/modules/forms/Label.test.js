import { renderToStaticMarkup } from 'react-dom/server';

import Label from './Label';

jest.mock('modules/i18n');

it('renders correctly', () => {
    const html = renderToStaticMarkup(
        <Label label="Name" className="label" htmlFor="input" />
    );
    expect(html).toMatchSnapshot();
});

it('renders with translation key', () => {
    const html = renderToStaticMarkup(<Label labelKey="label.default_name" />);
    expect(html).toMatchSnapshot();
});
