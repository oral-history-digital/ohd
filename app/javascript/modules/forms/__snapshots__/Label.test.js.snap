// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `"<div class="form-label label"><label class="FormLabel" for="input">Name</label></div>"`;

exports[`renders with translation key 1`] = `"<div class="form-label"><label class="FormLabel">label.default_name</label></div>"`;
